stravalib/model.py

```diff
@@ -926,10 +1242,35 @@ class SubscriptionCallback(
     hub_verify_token: Optional[str] = None
     hub_challenge: Optional[str] = None
 
+    # TODO: error: Required dynamic aliases disallowed  [pydantic-alias]
     class Config:
         alias_generator = lambda field_name: field_name.replace("hub_", "hub.")
 
-    def validate(self, verify_token=Subscription.VERIFY_TOKEN_DEFAULT):
+    # TODO: not sure if this docstring and typing is right- mypy is telling me
+    # there is an inconsistency -  Signature of "validate" incompatible with supertype "BaseModel"  [override]
+    def validate(
```

Comment:
if either of you happen to be able to help me understand what's happening here? it seems like it's somehow overriding pydantic's `BaseModel` looking at the errors and it does inherit from `DeprecatedSerializableMixin`.

`DeprecatedSerializableMixin(BaseModel)`

it seems like this is another override supporting legacy behavior. 

the full error is below
**************

1. it seems to want it to be a `@classmethod` - would that be a correct decorator for this object? and it doesn't like the mismatch in param names. 
```
Method "validate" overrides class "BaseModel" in an incompatible manner
  Base method is declared as a classmethod but override is not
  Parameter 2 name mismatch: base parameter is named "value", override parameter is named "verify_token"
  Return type mismatch: base method returns type "Model@validate", override returns type "None"
    Type "None" cannot be assigned to type "BaseModel*"Pylance[reportIncompatibleMethodOverride](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportIncompatibleMethodOverride)
main.py(683, 9): Overridden method
(method) def validate(
    self: Self@SubscriptionCallback,
    verify_token: str = Subscription.VERIFY_TOKEN_DEFAULT
) -> None
Validate the subscription's hub_verify_token against the provided token.

Parameters
verify_token : str
    The token to verify subscription If not provided, it uses the default VERIFY_TOKEN_DEFAULT.

Returns
None
    None if an assertion error is not raised

Raises
AssertionError
    If the hub_verify_token does not match the provided verify_token.
```
not sure if this is another spot to simply ignore the type check but seems like it might be if it's soon-to-be deprecated behavior??

Reply:
I find this one a bit problematic. We are here overwriting the `validate` method from `pydantic.BaseModel` this could cause some unexpected behavior. Ideally, this method should be renamed to something else. 
It's probably enough part of the internal of the library to allow us to change it?

I think the best solution here would be to put ignore here and open another issue to treat that problem

Reply:
> It's probably enough part of the internal of the library to allow us to change it?

This also came from the pre-pydantic version of stravalib, and I retained the name to be sure nothing was getting broken. Renaming this method looks like a relatively small risk of breaking stuff.